### PR TITLE
Allow filtering untargeted groups having no MS2 events

### PR DESCRIPTION
**Pull request by saifulbkhan**
This patch allows the user to filter untargeted groups based on whether it has any fragmentation events or not. This helps in narrowing down the number of groups they want to focus on in their DDA dataset.

### CODE REVIEW

**Comment by shubhra-agrawal** [comment]
@saifulbkhan Some conflicts here. Please rebase this and I'll test it tomorrow.

File: src/gui/mzroll/forms/peakdetectiondialog.ui [hunk 1]
```diff
@@ -202,31 +216,17 @@
...
-           <widget class="QDoubleSpinBox" name="mzMin">
-            <property name="maximum">
-             <double>999999999.000000000000000</double>
-            </property>
-           </widget>
-          </item>
-          <item row="1" column="2">
-           <widget class="QLabel" name="label_18">
+          <item row="2" column="0" colspan="2">
+           <widget class="QCheckBox" name="mustHaveMs2">
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/forms/peakdetectiondialog.ui]: @saifulbkhan If I check the box and reopen El-MAVEN, the box goes back to its unchecked state. Is that by design?

Comment (shubhra-agrawal) [reply src/gui/mzroll/forms/peakdetectiondialog.ui]: Okay, I see the advantage. If I load LCMS data in the next session, I would not want this box checked automatically. This is fine for now.

Comment (saifulbkhan) [reply src/gui/mzroll/forms/peakdetectiondialog.ui]: I do not recall making this decision explicitly. For it to save its checked state we might have to introduce a variable for this in the settings file as well. I can do that, if that is what we want.

Comment (shubhra-agrawal) [reply src/gui/mzroll/forms/peakdetectiondialog.ui]: @saifulbkhan Two ways to handle the situation I mentioned:
1. Disable/enable the checkbox based on LCMS/DDA data
2. Keep the behavior as it is but save the state in emDB

Comment (saifulbkhan) [reply src/gui/mzroll/forms/peakdetectiondialog.ui]: Got it. Will change it accordingly.

**Pull request merged**